qt/applications/workbench/workbench/plotting/figurewindow.py

```diff
@@ -121,12 +124,27 @@ def _plot_on_here(self, names, ax):
         if len(names) == 0:
             return
         # local import to avoid circular import with FigureManager
-        from mantidqt.plotting.functions import pcolormesh_from_names, plot_from_names
+        from mantidqt.plotting.functions import pcolormesh, plot_from_names, plot_surface, plot_wireframe
 
         fig = self._canvas.figure
         fig_type = figure_type(fig, ax)
         if fig_type == FigureType.Image:
-            pcolormesh_from_names(names, fig=fig)
```

Comment:
Do we now need to expand the `FigureType` enum? I feel we're bleeding out of the `figure_type` check such that we've got bits of code inside that and bits of code outside.

Can we also consider not using the `QuadMesh` check here too? The same axes will be created with a `pcolormesh` call in a script. Maybe consider the similar `Axes` check that is in `figuremanager`?